app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmissionController.scala

```diff
@@ -72,26 +73,31 @@ class RegistrationSubmissionController @Inject() (
     request: AuthorisedRequest[_]
   ): EitherT[Future, ResponseError, CreateEclSubscriptionResponsePayload] =
     for {
-      _              <- registrationValidationService
-                          .validateRegistration(EclRegistrationModel(registration, registrationAdditionalInfo))
-                          .asResponseError
-      now             = Instant.now().truncatedTo(ChronoUnit.SECONDS)
-      response       <- dmsService
-                          .submitToDms(registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml), now)
-                          .asResponseError
-      _               = if (appConfig.amendRegistrationNrsEnabled) {
-                          nrsService.submitToNrs(
-                            registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
+      _                <- registrationValidationService
+                            .validateRegistration(EclRegistrationModel(registration, registrationAdditionalInfo))
+                            .asResponseError
+      now               = Instant.now().truncatedTo(ChronoUnit.SECONDS)
+      registrationType <- valueOrError(registration.registrationType, "Registration type")
+      response         <- dmsService
+                            .submitToDms(
+                              registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml),
+                              now,
+                              registrationType
+                            )
+                            .asResponseError
+      _                 = if (appConfig.amendRegistrationNrsEnabled && registration.registrationType.contains(Amendment)) {
+                            nrsService.submitToNrs(
```

Comment:
Isn't this the equivalent of fire and forget if it's part of an assignment? It could potentially complete or fail after the request has returned to the user.

Reply:
Yes, this is how it has always been coded but I do also think it shouldn't be fire and forget